### ISSUE TITLE
Deleted Alert Policy causes unrecoverable error

### ISSUE THREAD

**Issue by ianrayns** [terraform-providers/terraform-provider-newrelic#57]
### Terraform Version
0.11.8

### Affected Resource(s)
- newrelic_alert_condition
- newrelic_infra_alert_condition
- newrelic_nrql_alert_condition
- newrelic_synthetics_alert_condition

### Expected Behavior
Alert conditions should be recreated if the policy does not exist

### Actual Behavior
Cannot progress past plan errors.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Create an alert policy and conditions
2. Delete alert policy via New Relic UI
3. Run `terraform plan`
4. Notice errors similar to `newrelic_nrql_alert_condition.master_count_high: Alerts policy {ID} could not be found.`

**Comment by stepankotyk** [comment]
having similar issue with dashboard. Seems tfstate is not refresh for deleted resources

**Comment by jhoward321** [comment]
Are there any updates on this?

**Comment by kalebwalton** [comment]
Same, we just ran into it.

**Comment by darrina** [comment]
What workarounds are there for this issue?  I'm seeing the same issue with Terraform v0.11.14 and NewRelic provider v1.5.0:

```
Creating process to run command [terraform11 init -no-color -reconfigure] in target folder [/tmp/tf/destroy-BIvEIEXgjDUjIPrE]

Initializing the backend...
Successfully configured the backend "s3"! Terraform will automatically
use this backend unless the backend configuration changes.
Initializing provider plugins...
- Checking for available provider plugins on https://releases.hashicorp.com...
- Downloading plugin for provider "aws" (1.60.0)...
- Downloading plugin for provider "template" (2.1.2)...
- Downloading plugin for provider "newrelic" (1.5.0)...
The following providers do not have any version constraints in configuration,
so the latest version was installed.
To prevent automatic upgrades to new major versions that may contain breaking
changes, it is recommended to add version = "..." constraints to the
corresponding provider blocks in configuration, with the constraint strings
suggested below.
* provider.template: version = "~> 2.1"
Terraform has been successfully initialized!
You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.
If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
Creating process to run command [terraform11 destroy -auto-approve -no-color] in target folder [/tmp/tf/destroy-BIvEIEXgjDUjIPrE]

Warning: data.aws_kms_secret.github: This data source will be removed in Terraform AWS provider version 2.0. Please see migration information available in: https://www.terraform.io/docs/providers/aws/guides/version-2-upgrade.html#data-source-aws_kms_secret
newrelic_alert_channel.slack: Refreshing state... (ID: 2707975)
newrelic_synthetics_monitor.engine_monitor: Refreshing state... (ID: 6b63628c-917b-42ee-9769-eafdc43e4c9d)
newrelic_alert_policy.alert_policy: Refreshing state... (ID: 481486)
alks_iamrole.lambda_role: Refreshing state... (ID: q-carvim-transform-role)
data.aws_caller_identity.home: Refreshing state...
data.aws_region.home: Refreshing state...
data.aws_kms_secret.github: Refreshing state...
data.aws_kms_secrets.github: Refreshing state...
aws_s3_bucket_object.sql_template_object: Refreshing state... (ID: q_carvim.xml)
aws_security_group.security_group: Refreshing state... (ID: sg-0c8ce8826a59755b3)
data.aws_region.this: Refreshing state...
data.aws_caller_identity.this: Refreshing state...
newrelic_synthetics_monitor_script.engine_monitor_script: Refreshing state... (ID: 6b63628c-917b-42ee-9769-eafdc43e4c9d)
aws_iam_role_policy.lambda_role_policy: Refreshing state... (ID: q-carvim-transform-role:lambda-policy)
aws_lambda_function.lambda_function: Refreshing state... (ID: q-carvim-transform)
newrelic_dashboard.carvim_receive_dashboard: Refreshing state... (ID: 930649)
aws_lambda_permission.allow_dlq_lambda: Refreshing state... (ID: AllowExecutionFromDlqLambda)
newrelic_alert_policy_channel.slack_channel: Refreshing state... (ID: 481486:2707975)
newrelic_infra_alert_condition.dlq_above_zero: Refreshing state... (ID: 481486:10998011)
newrelic_synthetics_alert_condition.engine_monitor_alert_condition: Refreshing state... (ID: 481486:10751383)

Error: Error refreshing state: 1 error(s) occurred:
* newrelic_synthetics_alert_condition.engine_monitor_alert_condition: newrelic_synthetics_alert_condition.engine_monitor_alert_condition: Alerts policy 481486 could not be found.
```

**Comment by ctrombley** [comment]
Fixed in #58.
